### DISCUSSION TITLE
Making RET related commands to stop deleting characters in a selection

### DISCUSSION THREAD

**Discussion by Doodler8888** [meow-edit/meow#696] [Q&A] [answered]
I want commands bound to RET to stop deleting characters if there is any selection active.
I've tried to make this advice:

```
(defun my/cancel-selection-before-ret (&rest _)
  "Cancel meow selection before executing RET binding."
  (when (and (meow-normal-mode-p)
             (region-active-p))
    (meow-cancel-selection)))

(dolist (func '(newline
                newline-and-indent
                org-return
                electric-newline-and-maybe-indent))
  (advice-add func :before #'my/cancel-selection-before-ret))
```

But it doesn't work for some reason, and i don't understand why.

**Comment by DogLooksGood** [comment]
Hey, 

If you press `RET` in meow normal state, it will just insert a newline and maybe indent after that. Nothing will be deleted, even with a region activated.

**Comment by Haxxflaxx** [comment, marked as answer]
Are you using  `delete-selection-mode` ?
If this is the case you should either turn it off globally and only enable it as needed i.e. activate it on `meow-insert-enter-hook` and disable it on `meow-insert-exit-hook`.
